corehq/apps/dump_reload/sql/load.py

```diff
@@ -160,11 +160,15 @@ def _group_objects_by_db(objects):
             except KeyError:
                 # in the case of foreign keys the serialized field name is the
                 # name of the foreign key attribute
-                field = [
-                    field for field in model._meta.fields
-                    if field.column == model.partition_attr
-                ][0]
-                partition_value = obj['fields'][field.name]
+                try:
+                    field = [
+                        field for field in model._meta.fields
+                        if field.column == model.partition_attr
+                    ][0]
+                    partition_value = obj['fields'][field.name]
+                except KeyError:
```

Comment:
Is the last line the only one that is expected to raise `KeyError`? Would this work?
```suggestion
                field = [
                    field for field in model._meta.fields
                    if field.column == model.partition_attr
                ][0]
                try:
                    partition_value = obj['fields'][field.name]
                except KeyError:
```